fdbclient/NativeAPI.actor.cpp

```diff
@@ -2850,14 +2826,80 @@ ACTOR Future<Void> getRangeStream( PromiseStream<Standalone<RangeResultRef>> res
 					wait( delay(CLIENT_KNOBS->WRONG_SHARD_SERVER_DELAY, info.taskID ));
 					break;
 				} else {
-					results.sendError(e);
+					results->sendError(e);
 					return Void();
 				}
 			}
 		}
 	}
 }
 
+ACTOR Future<Standalone<VectorRef<KeyRef>>> getRangeSplitPoints(Database cx, KeyRange keys, int64_t chunkSize);
+
+ACTOR Future<Void> getRangeStream(PromiseStream<RangeResult> _results, Database cx,
+                                  Reference<TransactionLogInfo> trLogInfo, Future<Version> fVersion, KeySelector begin,
+                                  KeySelector end, GetRangeLimits limits, Promise<std::pair<Key, Key>> conflictRange,
+                                  bool snapshot, bool reverse, TransactionInfo info, TagSet tags) {
+
+	state ParallelStream<RangeResult> results(_results, CLIENT_KNOBS->RANGESTREAM_CONCURRENT_FRAGMENTS);
+
+	// FIXME: better handling to disable row limits
+	ASSERT(!limits.hasRowLimit());
+	state Span span("NAPI:getRangeStream"_loc, info.spanID);
+
+	state Version version = wait(fVersion);
+	cx->validateVersion(version);
+
+	Future<Key> fb = resolveKey(cx, begin, version, info, tags);
+	state Future<Key> fe = resolveKey(cx, end, version, info, tags);
+
+	state Key b = wait(fb);
+	state Key e = wait(fe);
+
+	if (!snapshot) {
+		// FIXME: this conflict range is too large, and should be updated continously as results are returned
+		conflictRange.send(std::make_pair(std::min(b, Key(begin.getKey(), begin.arena())),
+		                                  std::max(e, Key(end.getKey(), end.arena()))));
+	}
+
+	if (b >= e) {
+		results.sendError(end_of_stream());
+		return Void();
+	}
+
+	state KeyRangeRef keys(b, e);
+
+	state Standalone<VectorRef<KeyRef>> splitPoints =
```

Comment:
It is probably a good idea to get the split points for each shard individually. When you do this call across multiple shard boundaries it will issues the requests to each shard sequentially, and it will pass the remaining bytes after the final key on one shard as a starting amount of bytes to the next shard. For our purposes we can only read from one shard at a time anyways